_overviews/collections-2.13/conversion-between-option-and-collection.md

```diff
@@ -0,0 +1,66 @@
+---
+layout: multipage-overview
+title: Conversion Between Option and Collection
+partof: collections-213
+overview-name: Collections
+
+num: 18
+previous-page: conversion-between-java-and-scala-collections
+
+permalink: /overviews/collections-2.13/:title.html
+---
+
+Is `Option` a proper Scala collection? To qualify as a collection at least two requirements must be satisfied 
+
+- can be iterated once or more than one time
+- can define [`fromSpecific`](https://github.com/scala/scala/blob/6c68c2825e893bb71d6dc78465ac8c6f415cbd93/src/library/scala/collection/Iterable.scala#L173) which turns one collection into another
+
+Now although an iterator of zero to one elements exists for `Option`, we cannot define 
+
+```
+def fromSpecific(coll: IterableOnce[A]): C
+```
+
+since implementation of `coll.headOption` would throw away elements when `coll` has more than one element. For this reason, starting [Scala 2.13](https://github.com/scala/scala/pull/8038), `Option` was made an `IterableOnce` but not an `Iterable`. To help clarify the difference between the two, mentally rename `Iterable` to `Collection` whilst `IterableOnce` becomes `Iterable`. Thus `Option` is iterable but not a collection. 
+
+One consequence of `Option` extending `IterableOnce` is that it can be [composed with other collections](https://stackoverflow.com/questions/4719592/composing-option-with-list-in-for-comprehension-gives-type-mismatch-depending-on) inside for-comprehension, for example
+
+```
+scala> for {
+     |   a <- Set(1)
+     |   b <- Option(41)
+     | } yield (a + b)
+val res0: scala.collection.immutable.Set[Int] = Set(42)
+``` 
+
+since `Set#flatMap` takes function to `IterableOnce`
+
+```
+scala> Set(1).flatMap
+def flatMap[B](f: Int => scala.collection.IterableOnce[B]): scala.collection.immutable.Set[B]
```

Comment:
This is not a valid Scala expression that users can copy-paste in a REPL, I think it is misleading to show it as such.